service/history/queue/cross_cluster_queue_processor_base.go

```diff
@@ -353,62 +349,57 @@ func (c *crossClusterQueueProcessorBase) readTasks(
 
 func (c *crossClusterQueueProcessorBase) pollTasks() []*types.CrossClusterTaskRequest {
 
-	// TODO: control the number of tasks returned
-	var result []*types.CrossClusterTaskRequest
-	for _, queueTask := range c.getTasks() {
-		crossClusterTask, ok := queueTask.(task.CrossClusterTask)
+	batchSize := 100
```

Comment:
This will be merged with https://github.com/uber/cadence/pull/4487 after it's landed and batch size will be controlled by a dynamic config